data/gym.js

```diff
@@ -43,23 +44,27 @@ const create = async (
     throw `Error: Create gym failed`;
   }
   const newId = insertInfo.insertedId.toString();
-  let user = await userFunctions.getByUserId(gymOwnerId)
+  let user = await userDataFunctions.getByUserId(gymOwnerId)
   user.gymsListForOwner.push(newId);
-  await userFunctions.update(gymOwnerId, user)
+  await userDataFunctions.update(gymOwnerId, user)
   const gym = await getByGymId(newId);
   return gym
 };
 
 const getByGymId = async (id) => {
+  id = id.trim(); // since on line 59 and 60 id is not trimmed yet; but also inside the remove function, id should is not trimmed and pass in.
   await validation.checkObjectId(id);
   const gymsDBConnection = await gymCollection();
-  const gymGet = await gymsDBConnection.findOne({_id: ObjectId(id.trim())});
+  const gymGet = await gymsDBConnection.findOne({ _id: ObjectId(id.trim()) });
   if (gymGet === null)
     throw [404, `ERROR: No gym exists with that id ${id.toString()}`];
   gymGet._id = gymGet._id.toString();
   return gymGet;
 };
 
+// TODO: we will need 
```

Comment:
Will this be needed in data/review or data/comment?

I thought this sorting logic would be in the router layer. The sorting logic will be presented in the router layer. The data layer will keep simple as CRUD.